File: src/main/java/com/github/magneticflux/aoc/year2015/day5/Problem2.java
package _2015.day5;

import java.io.FileNotFoundException;
import java.util.Iterator;
import java.util.LinkedList;
import java.util.List;
import java.util.Scanner;
import java.util.function.Predicate;

/**
 * Created by Mitchell on 12/18/2015.
 */
public class Problem2 {
    public static void main(String[] args) throws FileNotFoundException {
        Iterator<String> input = new Scanner(Problem2.class.getResourceAsStream("day5.txt"));
        List<String> strings = new LinkedList<>();
        input.forEachRemaining(strings::add);

        final String reg1 = ".*(\\w\\w)\\w*\\1.*";
        final String reg2 = ".*(\\w).\\1.*";

        Predicate<String> predicate = s -> {
            boolean hasPair = s.matches(reg1);
            boolean hasTrio = s.matches(reg2);
            return hasPair && hasTrio;
        };
        long num = strings.parallelStream().filter(predicate).count();
        System.out.println(num);
    }
}

File: src/main/kotlin/com/github/magneticflux/aoc/day4/Problem2.kt
package com.github.magneticflux.aoc.day4

import com.github.magneticflux.aoc.input
import java.util.*

object Problem2 {
    @JvmStatic
    fun main(args: Array<String>) {
        val input = Scanner(input())
        input.useDelimiter("\n")

        val count = input.iterator().asSequence().filter {
            val list = it.split(" ").map { String(it.toList().sorted().toCharArray()) }
            val set = list.toSet()
            set.size == list.size
        }.count()

        println("Count= $count")
    }
}